### PR TITLE
Added automation for NoSuchBucket exception for non existing bucket

### PR DESCRIPTION
**Pull request by vivekanandan-k-rh**
Signed-off-by: Vivekanandan K <vivk@redhat.com>

Added automation for NoSuchBucket exception for nonexisting bucket

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1667142

JIRA: https://issues.redhat.com/browse/RHCEPHQE-4466

### CODE REVIEW

**Review by udaysk23 — COMMENTED** [review]
@vivekanandan-k-rh Please attach the logs

**Comment by vivekanandan-k-rh** [comment]
@udaysk23 , Please find the log here: http://pastebin.test.redhat.com/1068249

File: rgw/v2/tests/s3_swift/test_acl_for_nonexisting_bucket.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            acls_set_status = s3_obj_acl.put(ACL="private")
+
+        except ClientError as e:
+            log.info(e.response)
+            response = HttpResponseParser(e.response)
+            if (
+                response.error["Code"] == "NoSuchBucket"
+                and e.response["Error"]["Code"] == "404"
+            ):
+                log.info("Setting acl for non existing bucket failed as expected")
```
Comment (udaysk23) [inline rgw/v2/tests/s3_swift/test_acl_for_nonexisting_bucket.py:64]: Looks like we are net getting correct value for `e.response["Error"]["Code"] == "404"`
@vivekanandan-k-rh could you please look into this once

Comment (vivekanandan-k-rh) [reply rgw/v2/tests/s3_swift/test_acl_for_nonexisting_bucket.py:64]: @udaysk23 , With the following HTTP response, it is confirmed that it s returning 404:
2022-07-27 10:46:48,515 INFO: http reponse:
{'Error': {'Code': 'NoSuchBucket', 'BucketName': 'nonexistingbucket'}, 'ResponseMetadata': {'RequestId': 'tx00000e5e6762d631aedf5-0062e14fd8-ac64-default', 'HostId': '', 'HTTPStatusCode': 404, 'HTTPHeaders': {'content-length': '227', 'x-amz-request-id': 'tx00000e5e6762d631aedf5-0062e14fd8-ac64-default', 'accept-ranges': 'bytes', 'content-type': 'application/xml', 'date': 'Wed, 27 Jul 2022 14:46:48 GMT', 'connection': 'Keep-Alive'}, 'RetryAttempts': 0}}

And also, if the response is not 404, test case will be failed

**Comment by vivekanandan-k-rh** [comment]
@udaysk23 , Please find the log for updated code: http://pastebin.test.redhat.com/1068269

File: rgw/v2/tests/s3_swift/test_acl_for_nonexisting_bucket.py (hunk 1 above)
Comment (udaysk23) [reply rgw/v2/tests/s3_swift/test_acl_for_nonexisting_bucket.py]: @vivekanandan-k-rh Thanks for changing condition parameters

**Review by udaysk23 — DISMISSED** [review]

File: rgw/v2/tests/s3_swift/test_acl_for_nonexisting_bucket.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
+"""
+Test NoSuchBucket error thrown for non existing bucket while applying ACL
+
+Usage: test_acl_for_nonexisting_bucket.py -c configs/<input-yaml>
```
[the remark this answers is not in the document]
Comment (vivekanandan-k-rh) [reply rgw/v2/tests/s3_swift/test_acl_for_nonexisting_bucket.py]: Done

File: rgw/v2/tests/s3_swift/test_acl_for_nonexisting_bucket.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        configure_logging(f_name=log_f_name, set_level=args.log_level.upper())
+        config = Config(yaml_file)
+        config.read()
+        test_exec(config)
+        test_info.success_status("test passed")
+        sys.exit(0)
+
+    except (RGWBaseException, Exception) as e:
+        log.info(e)
+        log.info(traceback.format_exc())
```
[the remark this answers is not in the document]
Comment (vivekanandan-k-rh) [reply rgw/v2/tests/s3_swift/test_acl_for_nonexisting_bucket.py]: Done

**Review by mkasturi18 — APPROVED** [review]

**Pull request merged**